go/vt/grpcclient/client_auth_static.go

```diff
@@ -59,12 +62,20 @@ func AppendStaticAuth(opts []grpc.DialOption) ([]grpc.DialOption, error) {
 	if *credsFile == "" {
 		return opts, nil
 	}
-	data, err := os.ReadFile(*credsFile)
-	if err != nil {
-		return nil, err
+
+	var err error
+	if credsData == nil {
+		var once sync.Once
+		once.Do(func() {
+			credsData, err = os.ReadFile(*credsFile)
```

Comment:
@tanjinx I suggest the `*StaticAuthClientCreds` struct is cached instead _(vs `[]byte`)_ so many goroutines don't need to repeat JSON unmarshalling the same `[]byte` data, ie: both `os.ReadFile(...)` and `json.Unmarshal(...)` happen within the `sync.Do(func() {...})`

Also we can defer this to another time, but I'd prefer if this can be reloaded with a `SIGHUP` signal. There *might* be re-usable code for this because it is supported for other credential files

Reply:
thanks Tim, good point, fixed it